core/src/components/textarea/textarea.md.scss

```diff
@@ -29,3 +32,18 @@
 .textarea-bottom .counter {
   letter-spacing: 0.0333333333em;
 }
+
+/**
+ * This makes the label sit above the textarea.
+ */
+:host(.textarea-label-placement-stacked) .label-text-wrapper,
+:host(.has-focus.textarea-label-placement-floating) .label-text-wrapper,
+:host(.has-value.textarea-label-placement-floating) .label-text-wrapper {
+  @include transform(translateY(6px), scale(#{$textarea-md-floating-label-scale}));
```

Comment:
The `6px` was calculated to allow for the same `9px` offset between the bottom of the label and the top of the textarea contents. If there is a better way to do this, I'm all for it. I unfortunately couldn't use the same `50%` scaling that `ion-input` uses, because input text vertically aligns text to the center of the control, textarea does not behave the same way with multiple lines of text. 

There may be style adjustments to smooth this out when doing the fill and outline styles as well.

|Legacy|Modern|
|---|---|
|![CleanShot 2022-12-20 at 14 26 37](https://user-images.githubusercontent.com/13732623/208750189-10844bce-9cd0-4a8f-9937-f86f123c700e.png)|![CleanShot 2022-12-20 at 14 26 14](https://user-images.githubusercontent.com/13732623/208750141-987e588c-d8e0-4445-826f-87ffe3d62218.png)|

Zooming works. Increasing font size beyond 23px does cause issues, but that is approximately the same font size that becomes problematic with floating labels in MDC: https://material-components.github.io/material-components-web-catalog/#/component/text-field (set `font-size` on `html` root).